### PR TITLE
Minor refactors and restructures in Concept

### PR DESCRIPTION
**Pull request by alexjpwalker**
## What is the goal of this PR?

To improve minor issues affecting code quality and help to align the code with client-nodejs

## What are the changes implemented in this PR?

Minor refactors and restructures in Concept

### CODE REVIEW

**Review by haikalpribadi — CHANGES_REQUESTED** [review]
We've now decided to name the `//concept/schema` package to be `//concept/logic` @alexjpwalker . So let's refactor `schema` to `logic` in the `//concept` package. cc: @flyingsilverfin @Phillammon

File: concept/schema/Rule.java [hunk 1]
```diff
@@ -17,15 +17,14 @@
...
+package grakn.client.concept.schema;
 
 import grakn.client.Grakn;
 import graql.lang.pattern.Pattern;
 
 import javax.annotation.CheckReturnValue;
-import javax.annotation.Nullable;
-import java.util.stream.Stream;
 
+// TODO: migrate to concept.schema package
```
Comment (flyingsilverfin) [inline concept/schema/Rule.java:27]: as haikal said, this package is: `concept.logic` is where this should now go

**Review by flyingsilverfin — CHANGES_REQUESTED** [review]

File: concept/schema/Rule.java (hunk 1 above)
Comment (alexjpwalker) [reply concept/schema/Rule.java]: done

File: .grabl/automation.yml [hunk 2]
```diff
@@ -79,7 +79,6 @@ build:
       dependencies: [build, build-dependency, test-behaviour]
     test-deployment-maven:
       machine: graknlabs-ubuntu-20.04
-      dependencies: [deploy-maven-snapshot]
```
Comment (haikalpribadi) [inline .grabl/automation.yml:82]: Why are we removing this dependency, @alexjpwalker ? Don't we need to make sure that `deploy-maven-snapshot` is successful before we test the maven deployment? Would be good to add comments to this PR btw.

**Review by haikalpribadi — APPROVED** [review]

**Pull request merged**